File: C++/implement_strStr.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if(needle.empty()) return 0;
        
        string tmp(needle);
        
        for(int i = 0; i < haystack.size(); i++){           
            if(haystack[i] == needle[0] && i+needle.size() <= haystack.size()) {
                for(int j = 0; j < needle.size(); j++) tmp[j] = haystack[i+j];
                if(needle == tmp) return i;
            }
        }
        return -1;
    }
};

File: Python3/implement_strStr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if not needle: return 0
        
        n_len = len(needle)
        
        for i in range(len(haystack)):
            if haystack[i] == needle[0] and (i+n_len <= len(haystack) and haystack[i:i+n_len] == needle): 
                return i
        return -1
